runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1961,24 +1960,24 @@ void J9::TransformUtil::removePotentialOSRPointHelperCalls(TR::Compilation* comp
 void J9::TransformUtil::prohibitOSROverRange(TR::Compilation* comp, TR::TreeTop* start, TR::TreeTop* end)
    {
    TR_ASSERT(start->getEnclosingBlock() == end->getEnclosingBlock(), "Does not support range across blocks");
+   TR_ASSERT(comp->supportsInduceOSR() && comp->isOSRTransitionTarget(TR::postExecutionOSR) && comp->getOSRMode() == TR::voluntaryOSR,
```

Comment:
Do we need something to check about if the OSR infrastructure has been removed already since this operation doesn't make sense in that world right?

Reply:
`comp->supportsInduceOSR()` does the check.